src/components/checkboxFilter/index.tsx

```diff
@@ -18,35 +18,64 @@ function CheckboxFilter<ItemKey extends string, FilterName extends string>({
   const hasGroups = items.some(
     (item) => (item.childCheckboxes ?? []).length > 0,
   );
+
+  const itemsPerColumn = Math.ceil(items.length / numberOfColumnsOnDesktop);
+
+  // Create an array to store items grouped by columns
+  const columns = Array.from(
+    { length: numberOfColumnsOnDesktop },
+    (_, columnIndex) => {
+      const columnStartIndex = columnIndex * itemsPerColumn;
+      const columnEndIndex = Math.min(
+        (columnIndex + 1) * itemsPerColumn,
+        items.length,
+      );
+      return [...items].slice(columnStartIndex, columnEndIndex);
+    },
+  );
+
   return (
     <TranslationProvider language={language} scopes={['checkboxFilter']}>
-      <Box
-        sx={{
-          columns: { md: numberOfColumnsOnDesktop, base: 1 },
-          columnRule: 'solid var(--chakra-colors-gray-100) 1px',
-          columnGap: 'var(--chakra-space-4xl)',
+      <Grid
+        gridTemplateColumns={{
+          base: '1fr',
+          md: `repeat(${numberOfColumnsOnDesktop}, 1fr)`,
         }}
+        gap="4xl"
       >
-        {items.map((item) => {
-          if (item.childCheckboxes && item.childCheckboxes.length > 0)
-            return (
-              <CheckboxGroupCollapsibleWithChildren
-                key={item.key}
-                item={item}
-                onApply={onApply}
-                onToggleCheckboxGroup={onToggleCheckboxGroup}
+        {columns.map((columnItems, columnIndex) => (
+          <Box key={columnIndex} data-testid="column" position="relative">
+            {/* Column separator */}
```

Comment:
Maybe we can create a component called `ColumnSeparator`? Or you can check if `<Divider />` can be used that we already have?